[Manutenção] Novo raspador para Coruripe-AL

LucasAlvws opened the pull request:
**AO ABRIR** uma *Pull Request* de um novo raspador (*spider*), marque com um `X` cada um dos items da checklist abaixo. Caso algum item não seja marcado, JUSTIFIQUE o motivo.

#### Layout do site publicador de diários oficiais
Marque apenas um dos itens a seguir:
- [ ] O *layout* não se parece com nenhum caso [da lista de *layouts* padrão](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/lista-sistemas-replicaveis.html)
- [ ] É um *layout* padrão e esta PR adiciona a spider base do padrão ao projeto junto com alguns municípios que fazem parte do padrão.
- [ ] É um *layout* padrão e todos os municípios adicionados usam a [classe de spider base](https://github.com/okfn-brasil/querido-diario/tree/main/data_collection/gazette/spiders/base) adequada para o padrão.

#### Código da(s) spider(s)
- [ ] O(s) raspador(es) adicionado(s) tem os [atributos de classe exigidos](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#UFMunicipioSpider).
- [ ] O(s) raspador(es) adicionado(s) cria(m) objetos do tipo Gazette coletando todos [os metadados necessários](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#Gazette).
- [ ] O atributo de classe [start_date](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#UFMunicipioSpider.start_date) foi preenchido com a data da edição de diário oficial mais antiga disponível no site.
- [ ] Explicitar o atributo de classe [end_date](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#UFMunicipioSpider.end_date) não se fez necessário.
- [ ] Não utilizo `custom_settings` em meu raspador.

#### Testes
- [ ] Uma coleta-teste **da última edição** foi feita. O arquivo de `.log` deste teste está anexado na PR.
- [ ] Uma coleta-teste **por intervalo arbitrário** foi feita. Os arquivos de `.log`e `.csv` deste teste estão anexados na PR.
- [ ] Uma coleta-teste **completa** foi feita. Os arquivos de `.log` e `.csv` deste teste estão anexados na PR.

#### Verificações
- [ ] Eu experimentei abrir alguns arquivos de diários oficiais coletados pelo meu raspador e verifiquei eles [conforme a documentação](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#diarios-oficiais-data-collection-data) não encontrando problemas.
- [ ] Eu verifiquei os arquivos `.csv` gerados pela minha coleta [conforme a documentação](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#tabela-da-coleta-arquivo-csv) não encontrando problemas.
- [ ] Eu verifiquei os arquivos de `.log` gerados pela minha coleta [conforme a documentação](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#log-arquivo-log) não encontrando problemas.

#### Descrição

<Descreva o seu Pull Request informando a issue (caso exista) que está sendo solucionada ou uma descrição do código apresentado>

LucasAlvws commented:
Uma coleta-teste por intervalo arbitrário
[al_coruripe_outubro2024.csv](https://github.com/user-attachments/files/17595276/al_coruripe_outubro2024.csv)
[log_al_coruripe_outubro2024.log](https://github.com/user-attachments/files/17595278/log_al_coruripe_outubro2024.log)

LucasAlvws commented:
Uma coleta-teste da última edição
[log_al_coruripe_last_edition.log](https://github.com/user-attachments/files/17595304/log_al_coruripe_last_edition.log)
[al_coruripe_last_edition.csv](https://github.com/user-attachments/files/17595306/al_coruripe_last_edition.csv)

LucasAlvws commented:
Rodei a consulta completa e houve alguns erros, mas não consegui entender exatamente o que são. Precisaria de uma ajuda para saber se posso resolve-los.
[al_coruripe.csv](https://github.com/user-attachments/files/17595364/al_coruripe.csv)
[log_al_coruripe.log](https://github.com/user-attachments/files/17595365/log_al_coruripe.log)

trevineju commented:
Boa, Lucas 

> Rodei a consulta completa e houve alguns erros, mas não consegui entender exatamente o que são. Precisaria de uma ajuda para saber se posso resolve-los. 

Conferiu essa seção da documentação? https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html#explorando-o-log

LucasAlvws commented:
Olá, após verificar a documetação e fiz mais alguns testes e modificações, mas não consigo fazer o download dessas duas edições
2024-11-03 22:26:48 [scrapy.downloadermiddlewares.retry] ERROR: Gave up retrying <GET https://diario.coruripe.al.gov.br/diarios/732/download> (failed 3 times): 500 Internal Server Error
2024-11-03 22:26:48 [scrapy.core.engine] DEBUG: Crawled (500) <GET https://diario.coruripe.al.gov.br/diarios/732/download> (referer: None)
2024-11-03 22:26:48 [scrapy.pipelines.files] WARNING: File (code: 500): Error downloading file from <GET https://diario.coruripe.al.gov.br/diarios/732/download> referred in <None>
2024-11-03 22:26:48 [scrapy.core.scraper] WARNING: Dropped: Validation failed!
{'_validation': defaultdict(<class 'list'>, {'files': ['Field too short']}),
 'date': '2024-07-04',
 'edition_number': '727',
 'file_urls': ['https://diario.coruripe.al.gov.br/diarios/732/download'],
 'files': [],
 'is_extra_edition': False,
 'power': 'executive',
 'scraped_at': '2024-11-04T01:22:19.541734Z',
 'territory_id': ''}
 
 2024-11-03 22:35:46 [scrapy.downloadermiddlewares.retry] ERROR: Gave up retrying <GET https://diario.coruripe.al.gov.br/diarios/128/download> (failed 3 times): 500 Internal Server Error
2024-11-03 22:35:46 [scrapy.core.engine] DEBUG: Crawled (500) <GET https://diario.coruripe.al.gov.br/diarios/128/download> (referer: None)
2024-11-03 22:35:46 [scrapy.pipelines.files] WARNING: File (code: 500): Error downloading file from <GET https://diario.coruripe.al.gov.br/diarios/128/download> referred in <None>
2024-11-03 22:35:46 [scrapy.core.scraper] WARNING: Dropped: Validation failed!
{'_validation': defaultdict(<class 'list'>, {'files': ['Field too short']}),
 'date': '2022-05-02',
 'edition_number': '124',
 'file_urls': ['https://diario.coruripe.al.gov.br/diarios/128/download'],
 'files': [],
 'is_extra_edition': False,
 'power': 'executive',
 'scraped_at': '2024-11-04T01:29:37.820665Z',
 'territory_id': ''}
 
 mesmo tentando fazer o download na mão eu não estou conseguindo. Vocês conseguiriam testar os links para mim?
 
 edição 124
![128](https://github.com/user-attachments/assets/d00893c4-4269-4f88-9a42-558d22fafd7d)

edição 727
![732](https://github.com/user-attachments/assets/95a295c1-509b-4b9f-a2d9-d627a7c23e0b)

firefueled commented on data_collection/gazette/spiders/al/al_coruripe.py:
```diff
@@ -0,0 +1,56 @@
+
+class AlCoruripeSpider(BaseGazetteSpider):
+    name = "al_coruripe"
+    TERRITORY_ID = ""
```
Este campo é obrigatório. Note que o campo vazio aqui resulta numa coluna vazia correspondente no csv.
Veja [na documentação](https://docs.queridodiario.ok.org.br/pt-br/latest/contribuindo/raspadores.html) tudo o que precisa fazer para garantir que o raspador seja integrado.

firefueled commented on data_collection/gazette/spiders/al/al_coruripe.py:
```diff
@@ -0,0 +1,56 @@
+
+    def start_requests(self):
+        url = f"{self.BASE_URL}/busca?term=&onde=tudo&data=qualquer&jornal=extra"
+        yield Request(url=url, method="GET", callback=self.parse)
```
Nota pro futuro: Os valores que definiu para `method` e `callback` são os padrões, então não precisava ter passado eles explicitamente :)

firefueled reviewed (changes requested):
Boa @LucasAlvws, 

Um erro 500 ao não conseguir baixar arquivos do site não necessariamente impede a integração do raspador.
Precisamos saber se a causa dos erros é o raspador ou o site que simplesmente falha para alguns diários.
Caso o problema seja no site, o raspador pode ser integrado caso todo o resto esteja OK.

Darei uma olhada nos erros 500 que encontrou para tentar entender. Enquanto isso, por favor considere os comentários que fiz.

firefueled commented:
Outra coisa,
O layout do site não se parece com nenhum da lista dos conhecidos e portanto deve marcar o item a seguir na descrição da sua PR.
"O layout não se parece com nenhum caso [da lista de layouts padrão]"
